demos/mqtt/mqtt_demo_mutual_auth/mqtt_demo_mutual_auth.c

```diff
@@ -0,0 +1,1608 @@
+ * overflow for 32 bit unsigned integer used for holding the timestamp.
+ *
+ */
+static uint32_t globalEntryTimeMs = 0U;
+
+/**
+ * @brief Packet Identifier generated when Subscribe request was sent to the broker;
+ * it is used to match received Subscribe ACK to the transmitted subscribe.
+ */
+static uint16_t globalSubscribePacketIdentifier = 0U;
+
+/**
+ * @brief Packet Identifier generated when Unsubscribe request was sent to the broker;
+ * it is used to match received Unsubscribe response to the transmitted unsubscribe
+ * request.
+ */
+static uint16_t globalUnsubscribePacketIdentifier = 0U;
+
+/**
+ * @brief Array to keep the outgoing publish messages.
+ * These stored outgoing publish messages are used to keep the messages until
+ * a successful ack is received.
+ */
+static PublishPackets_t outgoingPublishPackets[ MAX_OUTGOING_PUBLISHES ] = { 0 };
+
+/*-----------------------------------------------------------*/
+
+/**
+ * @brief Creates a TCP connection to the MQTT broker as specified by
+ * AWS_ENDPOINT and AWS_MQTT_PORT defined at the top of this file.
+ *
+ * @param[in] pServer Host name of server.
+ * @param[in] port Server port.
+ * @param[out] pTcpSocket Pointer to TCP socket file descriptor.
+ *
+ * @return EXIT_FAILURE on failure; EXIT_SUCCESS on success.
+ */
+static int connectToServer( const char * pServer,
+                            uint16_t port,
+                            int * pTcpSocket );
+
+/**
+ * @brief Reads credentials from the filesystem.
+ *
+ * Uses OpenSSL to import the root CA certificate, client certificate, and
+ * client certificate private key.
+ *
+ * @param[in] pSslContext Destination for the imported credentials.
+ * @param[in] pRootCaPath Path to the root CA certificate.
+ * @param[in] pClientCertPath Path to the client certificate.
+ * @param[in] pCertPrivateKeyPath Path to the client certificate private key.
+ *
+ * @return 1 if all credentials were successfully read; 0 otherwise.
+ */
+static int readCredentials( SSL_CTX * pSslContext,
+                            const char * pRootCaPath,
+                            const char * pClientCertPath,
+                            const char * pCertPrivateKeyPath );
+
+/**
+ * @brief Set up a TLS connection over an existing TCP connection.
+ *
+ * @param[in] tcpSocket Socket descriptor corresponding to the existing TCP connection.
+ * @param[in] pAlpnProtos List of ALPN protocols available to be negotiated.
+ * @param[in] alpnProtosLen Length of the ALPN protocols list.
+ * @param[out] pSslContext The output parameter to return the created SSL context.
+ *
+ * @return EXIT_FAILURE on failure; EXIT_SUCCESS on success.
+ */
+static int tlsSetup( int tcpSocket,
+                     const char * pAlpnProtos,
+                     size_t alpnProtosLen,
+                     SSL ** pSslContext );
+
+
+/**
+ * @brief The transport send function provided to the MQTT context.
+ *
+ * @param[in] pSslContext Pointer to SSL context.
```

Comment:
Would it be better to call it mqttNetworkContext as opposed to sslContext?

Reply:
Yes